src/digests.rs

```diff
@@ -282,6 +282,18 @@ pub trait MultihashDigest {
     ///
     /// Panics if the digest length is bigger than 2^32. This only happens for identity hasing.
     fn digest(&self, data: &[u8]) -> Multihash;
+
+    /// Digest input data.
+    ///
+    /// This method can be called repeatedly for use with streaming messages.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the digest length is bigger than 2^32. This only happens for identity hashing.
+    fn input(&mut self, data: &[u8]);
+
+    /// Retrieve the computed `Multihash`, consuming the hasher.
+    fn result(self) -> Multihash;
```

Comment:
should have result_reset and reset to allow easy reuse of the same instance

Reply:
Thought it would pollute the interface and take away from the original intention of the Digest structs, but np!

Reply:
Update: to do this, I think we'd need to replace  `blake2s_simd` with [`blake2`](https://crates.io/crates/blake2) as the current crate doesn't `impl Digest` or provide methods for resetting the hasher instance. `blake2` is written by the RustCrypto folks and also provides optional SIMD support, if required.

Reply:
We should not switch, the speed difference is pretty large. We can implement resetting manually for crates that don’t support it